x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorApiKeyIdAction.java

```diff
@@ -0,0 +1,152 @@
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.action;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.application.connector.Connector;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class UpdateConnectorApiKeyIdAction {
+
+    public static final String NAME = "cluster:admin/xpack/connector/update_api_key_id";
+    public static final ActionType<ConnectorUpdateActionResponse> INSTANCE = new ActionType<>(NAME);
+
+    private UpdateConnectorApiKeyIdAction() {/* no instances */}
+
+    public static class Request extends ActionRequest implements ToXContentObject {
+
+        private final String connectorId;
+
+        @Nullable
+        private final String apiKeyId;
+
+        @Nullable
+        private final String apiKeySecretId;
+
+        public Request(String connectorId, String apiKeyId, String apiKeySecretId) {
+            this.connectorId = connectorId;
+            this.apiKeyId = apiKeyId;
+            this.apiKeySecretId = apiKeySecretId;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.connectorId = in.readString();
+            this.apiKeyId = in.readOptionalString();
+            this.apiKeySecretId = in.readOptionalString();
+        }
+
+        public String getConnectorId() {
+            return connectorId;
+        }
+
+        public String getApiKeyId() {
+            return apiKeyId;
+        }
+
+        public String getApiKeySecretId() {
+            return apiKeySecretId;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(connectorId)) {
+                validationException = addValidationError("[connector_id] cannot be [null] or [\"\"].", validationException);
+            }
+
+            return validationException;
```

Comment:
I think that we should at least enforce to either apikeyid or apikeysecretid should be non-null for that request to pass the validation? What do you think about sth like: https://github.com/elastic/elasticsearch/blob/f048225d6241579e05681c74d2674b69e24db67e/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorNameAction.java#L86 ?

Reply:
I added this validation check and a yaml test to cover it